asmcomp/debug/dwarf_low/aranges_table.ml

```diff
@@ -0,0 +1,55 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                  Mark Shinwell, Jane Street Europe                     *)
+(*                                                                        *)
+(*   Copyright 2013--2018 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+module Int8 = Numbers.Int8
+module Int16 = Numbers.Int16
+
+type t = {
+  size : Dwarf_int.t;
+  values : Dwarf_value.t list;
+}
+
+let create ~start_of_code_symbol ~end_of_code_symbol
+      ~debug_info_label =
+  let module V = Dwarf_value in
+  let values = [
+    (* The initial length is inserted here by the code below. *)
+    V.Int16 (Int16.of_int_exn 2);  (* section version number *)
+    (* N.B. The following offset is to the compilation unit *header*, not
+       the compilation unit DIE. *)
+    V.Offset_into_debug_info debug_info_label;
+    V.Int8 (Int8.of_int_exn Arch.size_addr);
+    V.Int8 Int8.zero;  (* flat address space *)
+    (* end of header *)
+    V.Code_address_from_symbol start_of_code_symbol;
+    V.Code_address_from_symbol_diff
+      { upper = end_of_code_symbol; lower = start_of_code_symbol; };
+    V.Absolute_code_address Targetint.zero;
+    V.Absolute_code_address Targetint.zero;
```

Comment:
Ah, except that since we selected "flat address space" just above, then I think the terminating entry is only two zeroes (see the next sentence in the standard).  I have added some more comments in the code.

Reply:
Indeed.